app/views/devise/two_factor_authentication_setup/index.html.slim

```diff
@@ -10,5 +10,6 @@ h1.heading = t('devise.two_factor_authentication.two_factor_setup')
   p#2fa-description = t('devise.two_factor_authentication.otp_setup')
   = f.input :phone, as: :tel, label: 'Phone Number', required: true,
       input_html: { 'aria-describedby' => '2fa-description' }
+  = f.input :phone_sms_enabled, as: :boolean, label: 'SMS Enabled?', input_html: { checked: true }
```

Comment:
Or two buttons, like on AirBnB, `Verify via SMS` and `Verify via Call`
![airbnb](https://www.twilio.com/blog/wp-content/uploads/2015/06/sms-verification-featured-image.png)

Ideally, we wouldn't need to permanently store the user's preference in the DB. They would select it each time. This would depend on a change to the `two_factor_authentication` gem so that it doesn't automatically send a code right after Warden signs the user in. You should be able to send the code from the controller when you need to. The gem shouldn't force it upon you to send the code before the user reaches the controller.